src/wui/fieldaction.cc

```diff
@@ -324,6 +329,50 @@ void FieldActionWindow::init() {
 	warp_mouse_to_fastclick_panel();
 }
 
+static void gather_attributes(const Widelands::EditorGameBase& egbase,
```

Comment:
Performance: Do this once in `TribeDescr::process_productionsites` where we walk all the immovable attributes anyway.